src/components/app/data/utils.js

```diff
@@ -832,3 +832,21 @@ export function transformCourseMetadataByAllocatedCourseRunAssignments({
   }
   return courseMetadata;
 }
+
+/*
+ * A centralized helper to tell us if a given run is unrestricted for the given catalog.
+ */
+export function isRunUnrestricted({
+  restrictedRunsAllowed,
+  courseMetadata,
+  courseRunKey,
+  catalogUuid,
+}) {
+  const courseRunMetadata = courseMetadata.availableCourseRuns.find(r => r.contentKey === courseRunKey);
+  if (courseRunMetadata?.restrictionType === 'custom-b2b-enterprise') {
```

Comment:
nit: perhaps create a constant variable for `custom-b2b-enterprise` for re-use/portability?